### PR TITLE
Switch to moshi reflection adapter to avoid exposing a generated adapter in public API

### PR DESCRIPTION
**Pull request by jrodbx**
precursor to https://github.com/cashapp/paparazzi/pull/1264

cc: @TWiStErRob

### CODE REVIEW

File: paparazzi/build.gradle [hunk 1]
```diff
@@ -55,11 +55,10 @@ dependencies {
   api platform(libs.kotlin.bom)
   implementation libs.moshi.core
   implementation libs.moshi.adapters
+  implementation libs.moshi.kotlinReflect
   implementation libs.jcodec.core
   implementation libs.jcodec.javase
 
-  ksp libs.moshi.kotlinCodegen
```
Comment (TWiStErRob) [inline paparazzi/build.gradle:61]: hmm, bye-bye KSP Gradle Plugin for :paparazzi and :paparazzi-gradle-plugin?

Comment (jrodbx) [reply paparazzi/build.gradle:61]: i mean i guess i couldddddd use reflection only for `Snapshot` and codegen'd for the rest, but felt this was cleaner.  Any reason to reconsider?

Comment (TWiStErRob) [reply paparazzi/build.gradle:61]: That might be confusing to maintain, one way of serialization is probably enough.
Aren't all serialized classes public? It's the same problem for all.

Btw, my point was that this is the last `ksp` line, so the apply plugin can be removed. Which might totally remove KSP from the build.

File: paparazzi-gradle-plugin/build.gradle [hunk 2]
```diff
@@ -1,7 +1,6 @@
 apply plugin: 'org.jetbrains.kotlin.jvm'
 apply plugin: 'java-gradle-plugin'
 apply plugin: 'com.github.gmazzo.buildconfig'
-apply plugin: 'com.google.devtools.ksp'
```
Comment (JakeWharton) [inline paparazzi-gradle-plugin/build.gradle:4]: Bonus!

**Review by JakeWharton — APPROVED** [review]

**Pull request merged**